src/lib.rs

```diff
@@ -1,24 +1,24 @@
-extern "C" {
-    #![allow(improper_ctypes)] // we do not actually cross the FFI bound here
+#![allow(improper_ctypes)] // we do not actually cross the FFI bound here
 
+extern "C" {
```

Comment:
I think the `improper_ctypes` was on the wrong place (using `#!` instaed of plain `#`) -- can you move it back onto the function declaration in this extern block? Thanks!